vcfa/resource_vcfa_org_regional_networking.go

```diff
@@ -0,0 +1,167 @@
+				Required:    true,
+				Description: fmt.Sprintf(" %s", labelVcfaRegionalNetworkingSetting),
+			},
+			"org_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: fmt.Sprintf("Parent %s ID for %s", labelVcfaOrg, labelVcfaRegionalNetworkingSetting),
+			},
+			"provider_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: fmt.Sprintf("Parent %s ID for %s", labelVcfaProviderGateway, labelVcfaRegionalNetworkingSetting),
+			},
+			"region_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: fmt.Sprintf("Parent %s ID for %s", labelVcfaRegion, labelVcfaRegionalNetworkingSetting),
+			},
+			"edge_cluster_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: fmt.Sprintf("Backing %s ID for %s. Will be autoselected if not specified.", labelVcfaEdgeCluster, labelVcfaRegionalNetworkingSetting),
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: fmt.Sprintf("Status of %s", labelVcfaRegionalNetworkingSetting),
+			},
+		},
+	}
+}
+
+func resourceVcfaOrgRegionalNetworkingCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfaClient := meta.(*VCDClient)
+	c := crudConfig[*govcd.TmRegionalNetworkingSetting, types.TmRegionalNetworkingSetting]{
+		entityLabel:      labelVcfaRegionalNetworkingSetting,
+		getTypeFunc:      getTmRegionalNetworkingSettingType,
+		stateStoreFunc:   setTmRegionalNetworkingSettingData,
+		createFunc:       vcfaClient.CreateTmRegionalNetworkingSetting,
+		resourceReadFunc: resourceVcfaOrgRegionalNetworkingRead,
+	}
+	return createResource(ctx, d, meta, c)
+}
+
+func resourceVcfaOrgRegionalNetworkingUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfaClient := meta.(*VCDClient)
+	c := crudConfig[*govcd.TmRegionalNetworkingSetting, types.TmRegionalNetworkingSetting]{
+		entityLabel:      labelVcfaRegionalNetworkingSetting,
+		getTypeFunc:      getTmRegionalNetworkingSettingType,
+		getEntityFunc:    vcfaClient.GetTmRegionalNetworkingSettingById,
+		resourceReadFunc: resourceVcfaOrgRegionalNetworkingRead,
+	}
+
+	return updateResource(ctx, d, meta, c)
+}
+
+func resourceVcfaOrgRegionalNetworkingRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfaClient := meta.(*VCDClient)
+	c := crudConfig[*govcd.TmRegionalNetworkingSetting, types.TmRegionalNetworkingSetting]{
+		entityLabel:    labelVcfaRegionalNetworkingSetting,
+		getEntityFunc:  vcfaClient.GetTmRegionalNetworkingSettingById,
+		stateStoreFunc: setTmRegionalNetworkingSettingData,
+	}
+	return readResource(ctx, d, meta, c)
+}
+
+func resourceVcfaOrgRegionalNetworkingDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
I hit an issue with destroy, which was induced by a failing create at first. Steps to reproduce:
1. Get creation to fail (mine failed because I used same `name`, which was already used up in the underlying SDDC)
2. Regardless the failure, the networking still was created in UI (I could see it)
3. Then I tried to destroy everything with Terraform, but got an error.

In order to fix the destroy, I had to reimport the failed-to-create (during step 1) resource.

Reply:
I have changed the approach to use Async creation + ID recovery from failed task. The best we can do here is to store the ID with error. This will result in a tainted resource and one will be forced to recreated it on next apply. The good thing is that Terraform should be able to delete that stale/error'ed entity and create another one.

https://github.com/vmware/terraform-provider-vcfa/pull/18/files#diff-c20a383f64b5dfa2f1a5f478bce2f699e6bdbc6593ffb02d099ee15431a43451R71-R72